### PR TITLE
person class, student and also teacher class with their methods added to the project with individual files

### PR DESCRIPTION
**Pull request by MMhaidari**
# What is added new 
- [ ]  project created
- [ ] person class created
- [ ] student class created
- [ ] teacher class created
- [ ] student and teacher class have been inherited from person class 
- [ ] project is working fine

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    @parent_permission || of_age?
+  end
```
Comment (t-yanick) [inline person.rb:21]: - [ ] Great job working on this working on this milestone. As per project requirement, `can_use_services?` is to be a `public method`. Kindly make it public.
image

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @MMhaidari , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Presence of individual .rb file for each file
:heavy_check_mark: Gitflow used correctly
:heavy_check_mark: Work documented in a professional way
:heavy_check_mark: No linter errors

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MMhaidari** [comment]
# What is added new 
- [ ] can_use_services method added as a public method

**Review by ShadyShawkat — APPROVED** [review]
## Hi @MMhaidari, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**